Sources/SpeziAccessGuard/CodeViews/SetCodeView.swift

```diff
@@ -48,22 +51,34 @@ struct SetCodeView: View {
                     if !viewModel.setup {
                         state = .setCode
                     }
+                    
+                    // if the user is already unlocked, the user can directly set new passcode
+                    if !viewModel.locked {
+                        state = .setCode
+                    }
                 }
                 .transition(.opacity)
         case .setCode:
             VStack(spacing: 32) {
                 Text("SET_PASSCODE_PROMPT", bundle: .module)
                     .font(.title2)
                     .frame(maxWidth: .infinity)
-                CodeView(codeOption: $selectedCode) { code in
-                    guard selectedCode.verifyStructure(ofCode: code) else {
-                        errorMessage = String(localized: "PASSCODE_NOT_ACCORDING_TO_FORMAT", bundle: .module)
-                        return
-                    }
+                CodeView(codeOption: $selectedCode,
+                         toolbarButtonLabel: String(localized: "SET_PASSCODE_NEXT_BUTTON", bundle: .module)) {
+                    code, validationRes in
```

Comment:
Some suggestions on multiline formatting:
```suggestion
                CodeView(
                    codeOption: $selectedCode,
                     toolbarButtonLabel: String(localized: "SET_PASSCODE_NEXT_BUTTON", bundle: .module)
                 ) { code, validationRes in
```